Remove navigation bar appearance overrides

kean opened the pull request:
There are multiple problems caused by the legacy navigation bar appearance overrides.

- It [messes up](https://github.com/user-attachments/assets/913f3e5d-5686-4486-978a-4f270e554588) safe areas insets when using UIKit with SwiftUI 
- It breaks navigation bars in certain native screen, like photos picker – [#22333](https://github.com/wordpress-mobile/WordPress-iOS/issues/22333)
- It prevents the app from taking advantages of the new navigation view feature such as scroll edge appearance, and that makes the app feel outdated. The new appearance is especially beneficial for iPad.

Most of the new screens have been resetting these overrides to get the system behavior:

```swift
let viewController = UIHostingController(rootView: SubmitFeedbackView(presentingViewController: self, source: source))
viewController.configureDefaultNavigationBarAppearance()

let navigationController = UINavigationController(rootViewController: viewController)
navigationController.navigationBar.isTranslucent = true // Reset to default
```

This is not sustainable to keep doing it every screen, so it's time to bite a bullet and finally get rid of these and nearly all other global appearance overrides that break the assumptions about how the standard controls work. It will make the UI feel modern and will reduce the maintenance costs.

## Screenshots


| Before  | After |
| ------------- | ------------- |
|  <img width="320" alt="profile-after" src="https://github.com/user-attachments/assets/7e97f5a1-896b-4424-a677-5efee84a48b4"> | <img width="320" alt="Screenshot 2024-07-25 at 10 06 09 AM" src="https://github.com/user-attachments/assets/4eb78f0f-68f7-4be1-af94-39271bbf29fc"> |
| <img width="733" alt="ipad-before" src="https://github.com/user-attachments/assets/9e9e8b22-fe8f-4e00-9e84-1a58e6fd09ce"> | <img width="733" alt="ipad-after" src="https://github.com/user-attachments/assets/a768dfba-ef10-4596-a667-ec2e688e41e0"> |

## Regression Notes
1. Potential unintended areas of impact


2. What I did to test those areas of impact (or what existing automated tests I relied on)


3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

kean committed:
Remove navigation bar appearance overrides

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is larger than 500 lin...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

kean commented:
@jkmassel wdyt?

kean committed:
Remove more custom apperance code

kean committed:
Remove tooblar and tabbar appearance and isTranslucent changes breaking safe edeges

kean committed:
Remove more appearance workarounds

kean committed:
Remove preferredStatusBarStyle: UIStatusBarStyle

kean committed:
Remove more appearance customization

kean committed:
Fix notifications screen not extending below the tab bar

kean committed:
Remove remaining extendedLayoutIncludesOpaqueBars usages

kean committed:
Fix navigation bar in stats

kean committed:
Fix search bars

kean committed:
Rework TimeZoneSelectorViewController

kean committed:
Update TimeZoneSelectorViewController

kean commented:
There are a multiple screens that need to be updated to support the standard styles. I'm also using this is as on opportunity to phase out deprecated `.grouped` table view style.

## TimeZoneSelectorViewController

More / Site Settings / Time Zone

The correct table style to use on this screen is `.plain` with sections to ensure that section headers are sticky. I also updated insets to use standard system margins. The search primary button now says "Search" instead of "done".

| Before | After |
|------|-----|
| <img width="302" alt="before-1" src="https://github.com/user-attachments/assets/1146c092-43a7-4eba-916e-11c67f8f3d3b"> | <img width="302" alt="after-1" src="https://github.com/user-attachments/assets/82fbf766-db34-41fd-8407-27ebd7e2960d"> |
| <img width="302" alt="before-2" src="https://github.com/user-attachments/assets/788c8e26-8f65-49e8-bd2a-fa1ae3a52398"> | <img width="302" alt="after-2" src="https://github.com/user-attachments/assets/af9bdd05-0a5a-4670-b465-58894ebeab63"> |

## LanguageSelectorViewController

More / Site Settings / Language

Same treatment with `.plain` style and search bar displaying using `navigationItem`.

| Before | After |
|------|-----|
|<img width="302" alt="lang-before-1" src="https://github.com/user-attachments/assets/85778dbf-33a3-4b79-9100-f662d0b8dd7e"> |<img width="302" alt="lang-after-1" src="https://github.com/user-attachments/assets/0dcdb2df-8c5b-4ab0-b0a8-39c077dfcc2a">| 
| <img width="302" alt="lang-before-2" src="https://github.com/user-attachments/assets/b0d9750e-d294-4728-9c9f-6cefbc5f40b2">|<img width="302" alt="lang-after-2" src="https://github.com/user-attachments/assets/812774de-240c-438b-8b04-613965b7af68"> | 

## SiteTagsViewController

More / Site Settings / Tags

For this screen, I added two new SwiftUI-based APIs to replace now deprecated `DSButton` and `NoResultsViewController`.

`EmptyStateView` allows provides an easy way to create customizable empty states. Notice a new `.primary` button style designed in a "right" SwiftUI way. It also has a convenience `EmptyStateView.search()` factory method that can be used anywhere.

```swift
EmptyStateView {
    Label("You don't have any tags", systemImage: "tag")
} description: {
    Text("Tags created here can be easily added to new posts")
} actions: {
    Button {
        
    } label: {
        Text("Create a Tag")
    }
    .buttonStyle(.primary)
}
```

| Before | After |
|------|-----|
|<img width="300" alt="tags-before-1" src="https://github.com/user-attachments/assets/29edf8c6-3959-456e-b812-6672a686787a"> |<img width="300" alt="tags-after-1" src="https://github.com/user-attachments/assets/776f4d2f-441c-479e-8601-52f8ec0ce2ab"> | 
|<img width="300" alt="tags-before-2" src="https://github.com/user-attachments/assets/5e5890d1-a1a1-481a-80c9-9486ddd31745"> | <img width="300" alt="tags-after-2" src="https://github.com/user-attachments/assets/c8693857-cabf-40bf-b6f8-11011439f8c3">| 
|<img width="300" alt="tags-before-3" src="https://github.com/user-attachments/assets/b671306c-3ae0-4bbd-bedb-994ab74483e2"> | <img width="300" alt="tags-after-3" src="https://github.com/user-attachments/assets/99a0ee05-b1bb-4754-a3ce-7eb39d84599f">|
| <img width="300" alt="tags-before-4" src="https://github.com/user-attachments/assets/fc14a9ab-ef67-4213-ade7-e05596b3886d">| <img width="300" alt="tags-after-4" src="https://github.com/user-attachments/assets/f70aef75-68a5-4388-8b2a-7c1fe32aa6b2">| 

## DeleteSiteViewController

| Before | After |
|------|-----|
|<img width="300" alt="delete-site-before" src="https://github.com/user-attachments/assets/d25f0730-2dbc-4504-97cd-d4c62ad127ef"> | <img width="300" alt="delete-site-after" src="https://github.com/user-attachments/assets/65e64e21-2b21-45ba-9f3a-e736ebcd28ee">|

kean committed:
Update LanguageSelectorViewController

jkmassel commented:
Yep, we'll need to do this sooner or later - no objection to getting started on it now

kean committed:
Add UIHostingView

kean committed:
Rework SiteTagsViewController

kean committed:
Update SiteTagsViewController

kean committed:
/SiteTagsViewController

kean committed:
Nicer space used chart

kean committed:
Update DeleteSiteViewController design

kean committed:
Update SupportTableViewController design

kean committed:
Update StartOverViewController

kean committed:
Remove unused files

kean committed:
Update actionable cells design

kean committed:
Update StatsViewController design

kean committed:
Update StatsTrafficDatePickerView design:

kean committed:
Add tab bar tint color changes back

kean committed:
Fix layout of JetpackBannerWrapperViewController

kean committed:
Update PostStatsTableViewController design

kean committed:
Update ReferrerDetailsTableViewController

kean committed:
Update SiteStatsDetailTableViewController

kean committed:
Updte SiteStatsDetailTableViewController

kean committed:
Remove empty state create-post entry point - the screen has fab

kean committed:
Update BaseActivityListViewController design

kean committed:
Update acitivty log design

kean committed:
Update acitivty log design

kean committed:
Update PeopleViewController

kean committed:
Update theme browser

kean committed:
Update theme browser (2)

kean committed:
Update ReaderTagsTableViewController

kean committed:
Update notification details

kean committed:
Update All Domains

kean committed:
Update domain search

kean committed:
Cleanup WPStyleGuide

kean committed:
Remove more appearance overrides

kean committed:
Fix navigation bar colors in Aztec

kean committed:
Update ActivityDetailViewController

kean committed:
Remove unused password row from PastSettingsViewController

kean committed:
Update NotificationSettings

kean committed:
Update Activity filters

kean committed:
Update Aztec settings views

kean committed:
Update PluginList

kean committed:
Remove UIColor extensions

kean committed:
Remove some more UIColor extensions

kean committed:
Consolidate String extensions

kean committed:
Move some non-reusable code

kean committed:
Remove CustomHighlightButton

kean committed:
Remove unused LoadingStatusView

kean committed:
Remove unused FixedSizeImageView

kean committed:
Update restore site screen

kean committed:
Fix an issue with debugging web view

kean committed:
Update Share extension

kean committed:
Update NotificationSiteSubscriptionViewController

kean committed:
Fix MovedToJetpackViewController layout

kean committed:
Extract appearance from WordPressAppDelegate

kean committed:
Fix an issue with WebView debugging

kean committed:
Fix style of the backup icon

kean committed:
Fixes for iPad

crazytonyli commented:
Awesome work! The new UI looks great.

It's pretty difficult to review the code changes, due to large amount of deleted and moved code. Let me know if there is any specific code that you think worth a closer look, otherwise I'll browse the app and see if I can spot any UI issues.

crazytonyli commented:
View a notification detail, tap the site icon, the site icon in the site detail screen is misplaced.

| Before | After |
| ------ | ----- |
| ![IMG_5586](https://github.com/user-attachments/assets/d97b667c-5abe-49b8-b9da-c058e28753df) | ![IMG_5585](https://github.com/user-attachments/assets/7b70008b-ff14-484c-bd0e-552a02eaac64) |

crazytonyli commented:
View a notification, tap one of the tags/topics, the navigation bar push animation looks a bit strange.

https://github.com/user-attachments/assets/eea19598-09e8-457a-8c54-c6428c02b027

crazytonyli commented:
Tap the Reader tab, search something, go to the Blogs tab of the search result, the tab bar becomes transparent.

![IMG_5589](https://github.com/user-attachments/assets/b9b25f73-cc59-4e5c-8acb-51ec331b071e)

kean commented:
> It's pretty difficult to review the code changes, due to large amount of deleted and moved code. Let me know if there is any specific code that you think worth a closer look, otherwise I'll browse the app and see if I can spot any UI issues.

I tried to make only the most basic changes possible to make it unnecessary to review it line-by-line. As long as it works, it should be good. The only code I'm a bit worried about is in `SiteStatsDashboardViewController` where I added support for swiping between tabs. In hindsight, I should've left it out of this PR.

crazytonyli commented:
Found a bug in the new swiping features in stats. After dragging to the Traffic tab but returning to the original Insights tab, then tapping the Traffic tab, the page would stay on Insights tab. See the video recording below.

It's probably not obvious in the screen recording, all interactions are done by swiping, except the very last interaction to view Traffic tab which is done by tapping the tab.

https://github.com/user-attachments/assets/83b9057d-c317-4a6a-89c4-9e33aa76475b

crazytonyli commented on WordPress/WordPress.xcodeproj/xcshareddata/xcschemes/Jetpack.xcscheme:
```diff
@@ -121,11 +121,6 @@
-            key = "MallocScribble"
-            value = ""
-            isEnabled = "YES">
-         </AdditionalOption>
```
Just want to highlight this scheme change, which I think should be done: disabling Malloc Scribble diagnosis prevents WKWebView from crashing during debug sessions.

crazytonyli commented:
The navigation bar on Media detail screen is fully transparent.

![Screenshot 2024-07-31 at 1 38 03 PM](https://github.com/user-attachments/assets/0a332659-f6fb-4f1f-b7ec-50273da3bce7)

kean committed:
Fix reader issues

kean committed:
Remove paging in stats

kean committed:
Disable scroll edge appearance for SiteMediaPageViewController

kean commented:
> The navigation bar on Media detail screen is fully transparent.

Good catch. I don't remember testing scroll it vertically.

Fixed by reverting to the standard bar appearance using new  `WPStyleGuide.disableScrollEdgeAppearance`. It uses `UIPageViewController` and it doesn't seem to play well with scroll edge appearance. Ideally, it should be fixed differently, but I want to unblock the merge.

> Tap the Reader tab, search something, go to the Blogs tab of the search result, the tab bar becomes transparent.

Fixed, and fixed the details page. I'm not sure everything in Reader works. I'm wrapping up for today.
I'm a bit concerned about this part of the app. It needs to become much simpler to make it maintainable, but it's not an area I would invest much resources into.

> Found a bug in the new swiping features in stats.

I removed this change. As I mentioned earlier, it wasn't a good call to include it in this PR, but it will be a nice feature to add in the future.

crazytonyli commented:
The tab bar shows a different style when the Reader tab is selected.

| Reader | My Site |
| ------- | ------ |
| ![IMG_5592](https://github.com/user-attachments/assets/7ccc28bd-e5db-47a7-ab01-0eb6194622af) | ![IMG_5593](https://github.com/user-attachments/assets/fa074e31-e130-4b07-b22b-42e4cc20baa7) |

crazytonyli commented:
The navigation bar animation during pushing search screen in Reader looks odd.

https://github.com/user-attachments/assets/46e26f85-097d-4fd7-b934-8ff0540dc370

crazytonyli commented:
A modal presented from Reader has a transparent navigation bar.

Tap a reader post, scroll to the "More from ..." section at bottom of the post and tap a post, tap a tag/topic in the post in the presented modal, checkout the navigation bar in the pushed topic screen.

https://github.com/user-attachments/assets/dc660339-1496-4b17-b6e7-296ec6532846

kean committed:
Disable scroll edge appearance for the reader tabbar

kean committed:
Show related posts in the existing navigation stack

kean committed:
Improve Reader search

kean committed:
Fix bottom inset in reader stream

kean committed:
Fix background in ReaderTagStreamHeader

kean commented:
> The tab bar shows a different style when the Reader tab is selected.

I disabled scroll edge appearance for this tab for now, so it looks good now.

> The navigation bar animation during pushing search screen in Reader looks odd.

I fixed it and also fixed a production issue where the "recent searches" were covering the filter bar. It was apparently broken when the filter bar was added.

https://github.com/user-attachments/assets/75f2e2eb-acb9-4bd4-9798-0ab70d3d3557

> A modal presented from Reader has a transparent navigation bar.

I turned it into a push in the existing navigation stack because this is how every other navigation to the new topic/tag/site works in the Reader. I see no reason to show only related posts in a small sheet. We can't fix and maintain both types of presentation at the moment.

I also changes the background color of the stream to white/black, so there is no longer a gray gap when you use ptr.

I fixed the transition when selecting a topic in the main view as well. The issue was that the `ReaderStreamViewController` was configured to not extend below the top bars in storyboards.


@crazytonyli, I think it's an unacceptable state to get this PR merged and fix forward if there any issues. I want to make sure it's merged in trunk earlier so we don't hit big merge conflicts and the new code being developed uses the new global styles or lack thereof.

kean committed:
Fix animation when showing readere stream

crazytonyli commented:
> I think it's an unacceptable state to get this PR merged

@kean Just double checking, do you mean "acceptable"?

> [...] and fix forward if there any issues

Sure. I'll do a final walkthrough and will approve the PR if no major issue is found.

> The issue was that the ReaderStreamViewController was configured to not extend below the top bars in storyboards.

I noticed this issue on some other screens too, but didn't report them because they are small issues. But it'd be good to address when having a chance. Like the Me screen

![IMG_5598](https://github.com/user-attachments/assets/1b179e19-01ca-450b-8af3-61e8c2f9c9f4)

crazytonyli reviewed (approved):
All reported issues are addressed, and I couldn't find any major ones.

kean commented:
> @kean Just double checking, do you mean "acceptable"?

Yes 🤣 

> Sure. I'll do a final walkthrough and will approve the PR if no major issue is found.

I also plan to do more testing once the site picker changes and the newer iPad changes land.

> But it'd be good to address when having a chance. Like the Me screen

I added it to my list. I have a couple of tiny other things I noticed when testing.

> All reported issues are addressed, and I couldn't find any major ones.

🕺
